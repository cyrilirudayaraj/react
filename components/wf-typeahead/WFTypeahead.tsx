import React from 'react';
import { FormField, Typeahead, FormError } from '@athena/forge';
import { ErrorMessage } from 'formik';
import { Error } from '../error/Error';

const WFTypeahead = (props: any): JSX.Element => {
  return (
    <div className="typeahead-wrapper">
      <FormField
        inputAs={Typeahead}
        id={props.name}
        name={props.name}
        value={props.value}
        labelText={props.label}
        required
        onChange={props.onChange}
        onBlur={props.onBlur}
        alwaysRenderSuggestions={false}
        suggestions={props.suggestions}
        onSuggestionSelected={props.onSuggestionSelected}
        className="fe_u_margin--bottom-medium"
      />
      {props.showError === undefined ? (
        <ErrorMessage name={props.name} component={Error} />
      ) : props.showError ? (
        <div>
          <FormError>{props.errorMessage}</FormError>
        </div>
      ) : null}
    </div>
  );
};

export default WFTypeahead;
