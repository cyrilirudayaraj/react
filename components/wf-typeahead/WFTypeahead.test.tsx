import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import WFTypeahead from './WFTypeahead';

configure({ adapter: new Adapter() });
let wrapper: any;

const suggestions = ['apple', 'orange', 'banana'];

describe('snapshot test <WFTypeahead>', () => {
  it('should render as expected', () => {
    wrapper = shallow(
      <WFTypeahead label="Fruits" name="fruit" suggestions={suggestions} />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render with error', () => {
    wrapper = shallow(
      <WFTypeahead
        label="Fruits"
        name="fruit"
        suggestions={suggestions}
        showError={true}
      />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render with show error as false', () => {
    wrapper = shallow(
      <WFTypeahead
        label="Fruits"
        name="fruit"
        suggestions={suggestions}
        showError={false}
      />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
