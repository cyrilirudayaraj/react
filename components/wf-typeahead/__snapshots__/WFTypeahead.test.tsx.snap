// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test <WFTypeahead> should render as expected 1`] = `
"<div className=\\"typeahead-wrapper\\">
  <FormField inputAs={{...}} id=\\"fruit\\" name=\\"fruit\\" value={[undefined]} labelText=\\"Fruits\\" required={true} onChange={[undefined]} onBlur={[undefined]} alwaysRenderSuggestions={false} suggestions={{...}} onSuggestionSelected={[undefined]} className=\\"fe_u_margin--bottom-medium\\" />
  <FormikConnect(ErrorMessageImpl) name=\\"fruit\\" component={[Function: Error]} />
</div>"
`;

exports[`snapshot test <WFTypeahead> should render with error 1`] = `
"<div className=\\"typeahead-wrapper\\">
  <FormField inputAs={{...}} id=\\"fruit\\" name=\\"fruit\\" value={[undefined]} labelText=\\"Fruits\\" required={true} onChange={[undefined]} onBlur={[undefined]} alwaysRenderSuggestions={false} suggestions={{...}} onSuggestionSelected={[undefined]} className=\\"fe_u_margin--bottom-medium\\" />
  <div>
    <FormError alertType=\\"attention\\" />
  </div>
</div>"
`;

exports[`snapshot test <WFTypeahead> should render with show error as false 1`] = `
"<div className=\\"typeahead-wrapper\\">
  <FormField inputAs={{...}} id=\\"fruit\\" name=\\"fruit\\" value={[undefined]} labelText=\\"Fruits\\" required={true} onChange={[undefined]} onBlur={[undefined]} alwaysRenderSuggestions={false} suggestions={{...}} onSuggestionSelected={[undefined]} className=\\"fe_u_margin--bottom-medium\\" />
</div>"
`;
